Generator: Add full auroLabs docs (#95)
AlaskaAirlines/WC-Generator: docs/auroLabsMinors.md

    <div class="lightText">
      <p>Start the server to make sure that all dependencies are set up correctly.</p>
      <pre class="pre">  $ npm run serve</pre>
    </div>
  </auro-accordion>
  <auro-accordion id="newWork" chromeless noProfile>
    <span slot="trigger" class="trigger">6. Create an internal Github repo</span>
    <div class="lightText">
      <p>For Alaska employees only. Create a new <code>internal</code> repository with <auro-hyperlink target="_blank" href="https://github.com/AlaskaAirlines">github.com/AlaskaAirlines</auro-hyperlink>. Once completed, push your <code>main</code> branch to the remote repository.</p>
      <p>Please be sure to name the repo with <strong>auro</strong> to properly follow our open source repository guidelines. E.g. auro-[your component name].
      <p><strong>Randos will be deleted!</strong></p>
    </div>
  </auro-accordion>
</auro-accordion-group>

## Getting to work

Now that your install is setup and your repo is ready to go, let's get to work!

<auro-accordion-group>
  <auro-accordion id="newWork" chromeless noProfile>
    <span slot="trigger" class="trigger">1. Create your new feature branch</span>
    <div class="lightText">

>       <p>In your local repo, create a new working branch called <code>initial-build</code>, this will be the branch you will compare to <code>main</code> for the pull request later.</p>

Reviewer: ```suggestion
      <p>In your local repo, create a new working branch called <code>initial-build</code>. This will be the branch you will compare to <code>main</code> for the pull request later.</p>
```

    </div>
  </auro-accordion>
  <auro-accordion id="newWork" chromeless noProfile>
    <span slot="trigger" class="trigger">2. Edit the custom element file</span>
    <div class="lightText">

>       <p>60% of your work will happen in the custom element file, found at <code>./src/auro-[name].js</code>. From the generator there will be a scaffolding of a new custom element. We can't recommend enough that you review the <auro-hyperlink href="https://lit-element.polymer-project.org/guide/templates" target="_blank">LitElement docs</auro-hyperlink>. Many of the recommended best practices listed there will be seen in the generated file.</p>

Reviewer: ```suggestion
      <p>60% of your work will happen in the custom element file, found at <code>./src/auro-[name].js</code>. The generator will scaffold a new custom element. We can't recommend enough that you review the <auro-hyperlink href="https://lit-element.polymer-project.org/guide/templates" target="_blank">LitElement docs</auro-hyperlink>. Many of the recommended best practices listed there will be seen in the generated file.</p>
```

Author: ```suggestion
      <p>60% of your work will happen in the generated custom element file, found at <code>./src/auro-[name].js</code>. We can't recommend enough that you review the <auro-hyperlink href="https://lit-element.polymer-project.org/guide/templates" target="_blank">LitElement docs</auro-hyperlink>. Many of the recommended best practices listed there will be seen in the generated file.</p>
```

      <p>That's it. That's the magic.</p>
    </div>
  </auro-accordion>
  <auro-accordion id="newWork" chromeless noProfile>
    <span slot="trigger" class="trigger">4. Conventional commits</span>
    <div class="lightText">
      <p>Prior to making your first commit, please review <auro-hyperlink relative href="/contributing">Auro Design System Contributing Guidelines</auro-hyperlink>, specifically the section on <strong>Conventional Commits</strong>.</p>
      <p>Conventional Commits play a critical role in determining the next version release. The following template will help you construct the proper commit message.</p>
      <pre class="pre">  [type]([optional scope]): [description]</pre>
      <p>For example, let's say that you are working on a new feature supporting accessibility. Commit messages could be like the following. It is important to note that the <code>fix</code> commit is the only commit that will trigger a PATCH version release. The <code>docs</code> and <code>test</code> commits will not.</p>
      <pre class="pre">
  fix(a11y): update acronym to read out full word, e.g. SEA reads Seattle
  docs(a11y): address API changes in readme and automated docs

>   test(a11y): add new test for screenreader feature</pre>

Reviewer: ```suggestion
  test(a11y): add new test for screen reader feature</pre>
```

    </div>
  </auro-accordion>
  <auro-accordion id="newWork" chromeless noProfile>
    <span slot="trigger" class="trigger">5. Pre-commit hooks</span>
    <div class="lightText">
      <p>When committing code, you should see a series of pre-commit hooks confirming the stability of the code you are creating. If you do not see these hooks, then there is an issue with your install and you should consult with an Auro team member.</p>
    </div>
  </auro-accordion>
  <auro-accordion id="newWork" chromeless noProfile>
    <span slot="trigger" class="trigger">6. Amend commits to maintain a readable history</span>
    <div class="lightText">

>       <p>While Git history is helpful, in the early stages of a project not all history is helpful. It is considered best practice to <code>amend</code> previous commits if the work is closely related. Especially if it is a refactor of the previous commit. The following will add the updates recently completed and amend them to the previous commit without asking for a change in the commit message. Lastly, this is a change in history, so a force push to the remote is required.</p>

Reviewer: ```suggestion
      <p>While Git history is helpful, in the early stages of a project not all history is helpful. It is considered best practice to <code>amend</code> previous commits if the work is closely related, especially if it is a refactor of the previous commit. The following will add the updates recently completed and amend them to the previous commit without asking for a change in the commit message. Lastly, this is a change in history, so a force push to the remote is required.</p>
```
